core/common/src/main/java/alluxio/util/io/PathUtils.java

```diff
@@ -217,6 +217,17 @@ public static String temporaryFileName(long nonce, String path) {
   }
 
   /**
+   * @param path the path of the file, possibly temporary
+   * @return the permanent path of the file if it was temporary, or the original path if it was not
+   */
+  public static String getPermanentFileName(String path) {
+    if (isTemporaryFileName(path)) {
+      return path.substring(0, path.length() - 31);
```

Comment:
I tried to clean this up, but I couldn't find a nice way to link the String format with the regex expression we use to check if its a temp file.